### PR TITLE
docs(README): made readme better

### PR DESCRIPTION
**Pull request by pranavrao145**
Basically what the title says. Given that this plugin is constantly changing, the README was due for an update. This new README provides more information about the features of the plugin, configuration options, etc. etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
 
-**NEOVIM NIGHTLY**
-The plugin is tested on neovim nightly, please make sure you are on nightly
-before submitting an issue with the plugin.
+## Installation<a name="installation"></a>
 
-Plugins:
+### Requirements<a name="requirements"></a>
+
+- Neovim 0.6 Stable or 0.7 Nightly
```
Comment (teddylear) [inline README.md:35]: I think we should just say neovim nightly. Have already been bitten by some neovim upgrades with some issues being raised with pluign using new API for something.

File: README.md [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
-There will be some inevitable churn for the next week as I settle on the api!
+<div align="center">
 
-This is a WORK IN PROGRESS repo :)
+<h1>refactoring.nvim</h1>
+<h5>The Refactoring library based off the Refactoring book by Martin Fowler</h5>
 
-### Supported languages
+[![Lua](https://img.shields.io/badge/Lua-blue.svg?style=for-the-badge&logo=lua)](http://www.lua.org)
+[![Neovim](https://img.shields.io/badge/Neovim%200.6+-green.svg?style=for-the-badge&logo=neovim)](https://neovim.io)
```
Comment (teddylear) [inline README.md:7]: I think this might have to be updated to say neovim nightly too.

**Review by teddylear — CHANGES_REQUESTED** [review]
Overall looks WAY better than current README. One minor change about requiring neovim nightly.

**Comment by pranavrao145** [comment]
Copy that, I'll make those changes!

**Pull request merged**